src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,347 @@
+
+    /**
+     * When a ResumeAction command is recieved, the previously paused action shall resume and SHALL change the action’s state to
+     * Active.
+     *
+     * @param actionId The id of an action on which an action shall takes place.
+     * @return Returns a Success if an action took place successfully otherwise, suitable error.
+     */
+    virtual Protocols::InteractionModel::Status HandleResumeAction(uint16_t actionId, Optional<uint32_t> invokeId);
+
+    /**
+     * When an EnableAction command is recieved, it enables a certain action or automation. Afterwards, the action’s state SHALL be
+     * Active.
+     *
+     * @param actionId The id of an action on which an action shall takes place.
+     * @return Returns a Success if an action took place successfully otherwise, suitable error.
+     */
+    virtual Protocols::InteractionModel::Status HandleEnableAction(uint16_t actionId, Optional<uint32_t> invokeId);
+
+    /**
+     * When an EnableActionWithDuration command is recieved, it enables a certain action or automation, and SHALL change the
+     * action’s state to be Active. After the specified Duration, the action or automation will stop, and the action’s state SHALL
+     * change to Disabled.
+     *
+     * @param actionId The id of an action on which an action shall takes place.
+     * @param duration The time for which an action shall be in active state.
+     * @return Returns a Success if an action took place successfully otherwise, suitable error.
+     */
+    virtual Protocols::InteractionModel::Status HandleEnableActionWithDuration(uint16_t actionId, uint32_t duration,
+                                                                               Optional<uint32_t> invokeId);
+
+    /**
+     * When a DisableAction command is recieved, it disables a certain action or automation, and SHALL change the action’s state to
+     * Inactive.
+     *
+     * @param actionId The id of an action on which an action shall takes place.
+     * @return Returns a Success if an action took place successfully otherwise, suitable error.
+     */
+    virtual Protocols::InteractionModel::Status HandleDisableAction(uint16_t actionId, Optional<uint32_t> invokeId);
+
+    /**
+     * When a DisableActionWithDuration command is recieved, it disables a certain action or automation, and SHALL change the
+     * action’s state to Disabled. After the specified Duration, the action or automation will re-start, and the action’s state
+     * SHALL change to either Inactive or Active, depending on the actions.
+     *
+     * @param actionId The id of an action on which an action shall takes place.
+     * @param duration The time for which an action shall be in disable state.
+     * @return Returns a Success if an action took place successfully otherwise, suitable error.
+     */
+    virtual Protocols::InteractionModel::Status HandleDisableActionWithDuration(uint16_t actionId, uint32_t duration,
```

Comment:
So should some of these methods be pure virtual?  Where are they implemented?

Reply:
Right. Those are implemented in all-clusters-app.